HW01 - Jenkins container/Dockerfile

```diff
@@ -0,0 +1,48 @@
+# Use openjdk image as a source
+FROM openjdk:latest
+
+#================================
+#===== List of arguments ========
+
+# Directory with lenkins.war file
+ARG start_folder=/usr/bin/jenkins
+
+# Directory with Jenkins logs
+ARG log_folder=/var/log/jenkins
+
+# Folder with Jenkins data in user home folder
+ARG home_folder=/home/jenkins/.jenkins
+
+# Latest jenkins.war file
+ARG war_url=http://mirrors.jenkins.io/war-stable/latest/jenkins.war
+#================================
+
+# Add additional info to the image
+LABEL maintainer="Evgeny Minikh <evgeny.minih@gmail.com>"
+LABEL description="My first container with Jenkins"
+
+# Create necessary folders and add access rights for them
+RUN useradd -m jenkins && \
+	mkdir -p ${start_folder} && \
+	mkdir -p ${log_folder} && \
+	mkdir -p ${home_folder} && \
+	chown -R jenkins:jenkins ${home_folder} && \
+	chown -R jenkins:jenkins ${log_folder}
+
+# Copy build of Jenkins into the container
+ADD ${war_url} ${start_folder}/
+
+RUN chown -R jenkins:jenkins ${start_folder}
```

Comment:
Do you have any idea what this string exactly does?

Reply:
ADD ${war_url} ${start_folder} is used to download .war file from Jenkins site and put it into container (into directory from where we start Jenkins). I used ADD because COPY can't work with URL and archives. In previous step, when I created war file manually, I used COPY command, because .war file was stored on my laptop locally.

Second string adds access rigths to Jenkins start folder by changing folder's owner and owner group (recursively). It's because this folder was created with root user by default, but we want to use another user, so we have to change access rights to it. If we don't do it application will not started.